_rules/descriptive-labels-f0c5c5.md

```diff
@@ -0,0 +1,280 @@
+---
+id: f0c5c5
+name: Form field label is descriptive
+rule_type: atomic
+description: |
+  This rule checks that labels describe the purpose of form field elements.
+accessibility_requirements: 
+  wcag21:2.4.6: # Headings and labels (AA)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+- DOM Tree
+- CSS Styling
+authors:
+- Anne Thyme Nørregaard
+- Dagfinn Rømen
+- Geir Sindre Fossøy
+---
+
+## Applicability
+
+This rule applies to any element that:
+- has one of the following [semantic roles](#semantic-role): `checkbox`, `combobox` (`select` elements), `listbox`, `menuitemcheckbox`, `menuitemradio`, `radio`, `searchbox`, `slider`, `spinbutton`, `switch` and `textbox`, and
+- is [visible](#visible) or [included in the accessibility tree](#included-in-the-accessibility-tree), and 
+- has an [accessible name](#accessible-name) or is the [labeled control](https://www.w3.org/TR/html/sec-forms.html#labeled-control) of a [visible](#visible) `label` element.
+
+**Note**: The list of applicable [semantic roles](#semantic-roles) is derived by taking all the [ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1/) roles that:
+- inherit from the [abstract](https://www.w3.org/TR/wai-aria/#abstract_roles) `input` or `select` role, and 
+- do not have a [required context](https://www.w3.org/TR/wai-aria/#scope) role that itself inherits from one of those roles.
+
+**Note:** The `option` role is not part of the list of applicable roles, because it has a required context role that inherits from the `select` role. Furthermore, `option` does not meet the definition of a [User interface component](https://www.w3.org/TR/WCAG21/#dfn-user-interface-components). This means [WCAG 2.1](https://www.w3.org/TR/WCAG21/) does not require it to have an accessible name.
+
+**Note**: This rule is a partial check for WCAG 2.1 success criterion 2.4.6, which applies to all labels. "Label" in WCAG is used in its general sense and includes text or other components with a text alternative that is presented to a user to identify a component within web content.
```

Comment:
I think it may be good to point out that "label" means the `label` element in this rule while it means something else in WCAG. I agree that stating this rule is a partial check is no the way we usually go and thus should be skipped.